cli/pcluster/config/validators.py

```diff
@@ -1066,3 +1069,52 @@ def _get_efa_enabled_instance_types(errors):
         )
 
     return instance_types
+
+
+def fsx_lustre_backup_validator(param_key, param_value, pcluster_config):
+    errors = []
+    warnings = []
+
+    try:
+        backups = boto3.client("fsx").describe_backups(BackupIds=[param_value]).get("Backups")
+        if backups is None or len(backups) <= 0:
+            errors.append("Backup '{0}' does not exist.".format(param_value))
+            return errors, warnings
+
+        backup = backups[0]
+        fsx_section = pcluster_config.get_section("fsx")
+        backup_deployment_type = backup.get("FileSystem").get("LustreConfiguration").get("DeploymentType")
+        deployment_type = fsx_section.get_param_value("deployment_type")
```

Comment:
In my opinion instead of requiring customers to copy the correct values for `deployment_type`, `per_unit_storage_throughput`, and `storage_capacity` the parameters into the config file, we should just populate those parameters using values read from the DescribeBackup API when restoring a backup.

Reply:
Sorry my bad. I think FSx already does inherit deployment_type, per_unit_storage_throughput and storage_capacity from the backup. So we need these checks only when the customer explicitly configures these parameters. There should be None value check for each of these parameters. If its not None, only then we will check for equality.

Reply:
In my opinion, if a backup to restore from is specified, then specifying any of the parameters associated with that  backup should be a validation error.